bootstrap/api/v1alpha1/rke2config_types.go

```diff
@@ -18,47 +18,624 @@ package v1alpha1
+	// +optional
+	ClusterDNS string `json:"clusterDNS,omitempty"`
+
+	// ClusterDomain Cluster Domain (default: "cluster.local")
+	// +optional
+	ClusterDomain string `json:"clusterDomain,omitempty"`
+
+	// WriteKubeconfig gives the path to which kubeconfig file for admin client will be written
+	// +optional
+	WriteKubeconfig string `json:"writeKubeconfig,omitempty"`
+
+	// WriteKubeconfigMode Writes kubeconfig with this mode
+	// +optional
+	WriteKubeconfigMode string `json:"writeKubeconfigMode,omitempty"`
+
+	// EtcdExposeMetrics Expose etcd metrics to client interface. (Default false)
+	// +optional
+	EtcdExposeMetrics string `json:"etcdExposeMetrics,omitempty"`
+
+	// EtcdBackupConfig defines how RKE2 will snapshot ETCD: target storage, schedule, etc.
+	//+optional
+	EtcdBackupConfig EtcdBackupConfig `json:"etcdBackupConfig,omitempty"`
+
+	// Disable Do not deploy packaged components and delete any deployed components (valid items: rke2-coredns, rke2-ingress-nginx, rke2-metrics-server)
+	// +optional
+	Disable []DisabledItem `json:"disable,omitempty"`
+
+	// DisableScheduler Disable Kubernetes default scheduler
+	// +optional
+	DisableScheduler string `json:"disable-scheduler,omitempty"`
+
+	// DisableCloudController Disable rke2 default cloud controller manager
+	// +optional
+	DisableCloudController string `json:"disableCloudController,omitempty"`
+
+	// DisableKubeProxy Disable running kube-proxy
+	// +optional
+	DisableKubeProxy string `json:"disableKubeProxy,omitempty"`
+
+	// NodeNamePrefix Prefix to be used by CAPI to generate the Node name
+	// +optional
+	NodeNamePrefix string `json:"nodeNamePrefix,omitempty"`
+
+	// NodeLabel Registering and starting kubelet with set of labels
+	// +optional
+	NodeLabels []string `json:"nodeLabel,omitempty"`
+
+	// NodeTaint Registering kubelet with set of taints
+	// +optional
+	NodeTaints []string `json:"nodeTaint,omitempty"`
+
+	// TODO: Remove ImageCredentialProviderBinDir when implementation is understood.
+	// ImageCredentialProviderBinDir The path to the directory where credential provider plugin binaries are located (default: "/var/lib/rancher/credentialprovider/bin")
+	// +optional
+	//ImageCredentialProviderBinDir string `json:"imageCredentialProviderBinDir,omitempty"`
+
+	// ImageCredentialProviderConfigMap is a reference to the ConfigMap that contains credential provider plugin config
+	// The configMap should contain a YAML file content + a Path to the Binaries for Credential Provider.
+	// +optional
+	ImageCredentialProviderConfigMap ConfigMapRef `json:"imageCredentialProviderConfigMap,omitempty"`
+
+	// TODO: Remove ContainerRuntimeEndpoint since this feature will probably not be offered by CAPI Bootstrap provider?
+
+	// ContainerRuntimeEndpoint Disable embedded containerd and use alternative CRI implementation
+	// +optional
+	ContainerRuntimeEndpoint string `json:"containerRuntimeEndpoint,omitempty"`
+
+	// Snapshotter Override default containerd snapshotter (default: "overlayfs")
```

Comment:
whare are the other options besides `overlayfs `? if they are limited, we can make an enum

Reply:
The list is available [here](https://github.com/containerd/containerd/blob/f033f6ff85ce397202ee69de22b4e3a4bf4093f5/docs/snapshotters/README.md#snapshotters), but the more I think about this, the more I am convinced we should not expose this option to users. 

I think it is way too advanced, and might cause more headaches than anything else. 

What do you think?

Reply:
yeah, we can remove it and add it back if someone asks for support.

Reply:
Done!